frontend/gateway/gateway.go

```diff
@@ -64,6 +64,7 @@ func (gf *gatewayFrontend) Solve(ctx context.Context, llbBridge frontend.Fronten
 	_, isDevel := opts[keyDevel]
 	var img ocispec.Image
 	var rootFS cache.ImmutableRef
+	readonly := true // XXX could this be set to false for all cases below?
```

Comment:
I cannot at all remember what I meant by this comment, since setting the containers to readonly is surely more desirable when possible.

Using readonly doesn't appear to have caused any problems with the test suite, but I don't know how good the test coverage is here (esp. for `gateway-devel` mode). @tonistiigi shall I drop the comment or switch to writable rootfs do you think?